src/client/sky.cpp

```diff
@@ -253,34 +253,10 @@ void Sky::render()
 			driver->setMaterial(m_materials[1]);
 			for (u32 j = 0; j < 4; j++) {
 				video::SColor c = cloudyfogcolor.getInterpolated(m_skycolor, 0.45);
-				vertices[0] = video::S3DVertex(-1, 0.08, -1, 0, 0, 1, c, t, t);
-				vertices[1] = video::S3DVertex( 1, 0.08, -1, 0, 0, 1, c, o, t);
-				vertices[2] = video::S3DVertex( 1, 0.12, -1, 0, 0, 1, c, o, o);
-				vertices[3] = video::S3DVertex(-1, 0.12, -1, 0, 0, 1, c, t, o);
-				for (video::S3DVertex &vertex : vertices) {
-					if (j == 0)
-						// Don't switch
-						{}
-					else if (j == 1)
-						// Switch from -Z (south) to +X (east)
-						vertex.Pos.rotateXZBy(90);
-					else if (j == 2)
-						// Switch from -Z (south) to -X (west)
-						vertex.Pos.rotateXZBy(-90);
-					else
-						// Switch from -Z (south) to +Z (north)
-						vertex.Pos.rotateXZBy(-180);
-				}
-				driver->drawIndexedTriangleFan(&vertices[0], 4, indices, 2);
-			}
-
-			// Draw far cloudy fog thing at and below all horizons
```

Comment:
Careful ... there may be a reason why this farcloudyfogthing is drawn down to -1.0, even though it is later overlaid by another layer at 'Draw far cloudy fog thing below all horizons in front of sun, moon and stars.'.
There is often a reason for the way things are.
It is possible that you cannot remove a drawing stage like this.

Reply:
There often is, but it doesn't seem to be the case here, both 'things' are drawn under the same condition.

I don't know how well gpus handle fill rate these days, but drawing two large, opaque surfaces on top of one another probably isn't a great idea.

Reply:
I agree with @TheTermos - if it can't be justified now, it shouldn't be left there. If it's overlaid, there's room for optimization, and should be optimized.

Reply:
Yes, it is probably ok.
appgurueu, i am only saying 'be careful', i am not opposing this change.